source/puppet/4.4/reference/static_catalogs.md

```diff
@@ -0,0 +1,130 @@
+
+When applying a file resource from a static catalog, an agent first checks the catalog for that file's metadata. If it finds some, Puppet uses the metadata to call the [`static_file_content`][] API endpoint on the Puppet master and retrieve the `code_content`. If the catalog doesn't contain metadata for the resource, Puppet does what it's always done: generate the metadata itself, then request the resource's file from the master in the file's current state.
+
+### Configuring `code_id` and the `static_file_content` endpoint
+
+When requesting the file's content from a static catalog's metadata, the Puppet agent passes the file's path, the catalog's `code_id`, and the requested environment to Puppet Server's new [`static_file_content`][] API endpoint, and the endpoint returns the appropriate version of the file's contents as the `code_content`.
+
+> **Puppet Enterprise note:** If [file sync][] is enabled on future versions of Puppet Enterprise that include Puppet Server 2.3, Puppet Server automatically populates the `code_id` and serves the `code_content` out of the box. You don't need to read any further to configure static catalogs --- they're already configured. However, if you've disabled file sync (for instance, to use r10k instead of Code Manager) and want to use static catalogs, you'll need to read on for information about creating the necessary scripts and configuring Puppet Server.
+
+If static catalogs are enabled but none of the relevant Puppet Server settings are configured, the `code_id` parameter defaults to a null value and the `static_file_content` API endpoint always returns the latest content. To populate the `code_id` with a more useful identifier and have the `static_file_content` endpoint return the appropriate version of the file's content, you must specify scripts or commands that provide Puppet with the right information.
+
+Puppet Server locates these commands via the `code-id-command` and `code-content-command` settings in Puppet Server's [`puppetserver.conf`][] file. Puppet Server runs the `code-id-command` each time it compiles a static catalog, and it runs the `code-content-command` each time an agent requests file contents from the `static_file_content` endpoint.
+
+> **Puppet Enterprise note:** You can set `code-id-command` and `code-content-command` in the PE console by changing the `code_id_command` and `code_content_command` parameters. Note that these only take effect if [file sync][] is disabled by setting the `file_sync_enabled` console parameter to false.
```

Comment:
Between a rock and a hard place: Ideally this would link back to the PE docs covering this subject, but they won't exist when 2.3.0's released. The disclaimer would be moot when 2.3.0 comes out anyway, and would be covered by the PE docs when PE includes this feature. There'll be a follow-on PR once that's the case.

Alternatively, we can just pull all the PE-specific stuff from this PR and put it in that follow-on PR.